bin/parallel.py

```diff
@@ -77,14 +80,17 @@ def _interrupt_handler(self, sig, frame):
     # Add one task.
     def put(self, task):
         if not self.num_threads:
-            self.f(task)
-            return
+            # no task should be added after .done() was called
+            assert not self.finish
+            # no task will be handled after self.abort
+            if not self.abort:
+                self.f(task)
+                return
```

Comment:
Move this return one level up? With `self.abort == True` we also need to return here.